app/graphql/types/system.rb

```diff
@@ -51,11 +51,15 @@ def profile_names
     end
 
     def rules_passed(args = {})
-      object.rules_passed(::Profile.find_by(id: args[:profile_id]))
+      ::RecordLoader.for(::Profile).load(args[:profile_id]).then do |profile|
+        object.rules_passed(profile)
```

Comment:
No changes here - this just batches the `Profile` find call in case you are finding multiple profiles, e.g: if you're getting all profiles for a host.